fastlane/lib/fastlane/actions/xcodes.rb

```diff
@@ -0,0 +1,150 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      XCODES_XCODE_PATH = :XCODES_XCODE_PATH
+    end
+
+    class XcodesAction < Action
+      def self.run(params)
+        # Provide xcodes with the necessary env vars
+        ENV["XCODES_USERNAME"] = params[:username]
+        ENV["XCODES_PASSWORD"] = params[:password]
+        binary = params[:binary_path]
+        select_for_current_build_only = params[:select_for_current_build_only]
+        version = params[:version]
+        if params[:update_list] && !select_for_current_build_only
+          command = []
+          command << binary
+          command << "update"
+          shell_command = command.join(' ')
+          UI.message("Available versions:")
+          Actions.sh(shell_command)
+        end
+
+        if (xcodes_args = params[:xcodes_args])
+          command = []
+          command << binary
+          command << xcodes_args
+          shell_command = command.join(' ')
+          Actions.sh(shell_command)
+        elsif !select_for_current_build_only
+          command = []
+          command << binary
+          command << "install"
+          command << "'#{version}'"
+          shell_command = command.join(' ')
+          Actions.sh(shell_command)
+        end
+
+        command = []
+        command << binary
+        command << "installed"
```

Comment:
This doesn't work (anymore?):

```plaintext
[21:10:54]: --------------------
[21:10:54]: --- Step: xcodes ---
[21:10:54]: --------------------
[21:10:54]: $ /opt/homebrew/bin/xcodes installed '13.3'
[21:10:54]: ▸ Error: Unexpected argument '13.3'
[21:10:54]: ▸ Usage: xcodes installed [--directory <directory>] [--color] [--no-color]
[21:10:54]: ▸ See 'xcodes installed --help' for more information.
+------------------+------------------+
|            Lane Context             |
+------------------+------------------+
| DEFAULT_PLATFORM | ios              |
| PLATFORM_NAME    | ios              |
| LANE_NAME        | ios ensure_xcode |
+------------------+------------------+
[21:10:54]: Exit status of command '/opt/homebrew/bin/xcodes installed '13.3'' was 64 instead of 0.
Error: Unexpected argument '13.3'
Usage: xcodes installed [--directory <directory>] [--color] [--no-color]
  See 'xcodes installed --help' for more information.
```

I'm using `xcodes` version `1.0.0`.

Reply:
It doesn't work 'yet' 🙈 This PR this currently blocked by the release of the next version of `xcodes` 😬 as I stated in the PR description:

>This PR depends on https://github.com/RobotsAndPencils/xcodes/pull/220 merge & release so the installed option with specified version is available.

It has been merged, but not yet released. I've contacted the maintainer of xcodes (@MattKiazyk 🙌 ) and he said he doesn't have a release cadence, so yeah we need to wait for the next release to see this feature available

Any suggestions to enhance this @revolter ? 🙇

Reply:
> This PR depends on https://github.com/RobotsAndPencils/xcodes/pull/220 merge & release so the installed option with specified version is available.

Oh, I missed the "release" part 🤦🏻‍♂️ 😂

> Any suggestions to enhance this @revolter?

Enhance what?

Reply:
@revolter nevermind, I thought of adding a version check but there wouldn't be a great way to implement the same functionality before that feature gets released on `xcodes`. Let's just wait 🙇

Reply:
I ended up adding this version check (for v1.1.0) just for validation sake (it crashes if <1.1.0) so users are aware of this limitation 👍